python-api/src/main/python/livy/job_handle.py

```diff
@@ -247,8 +235,8 @@ def start(self):
             if not self.is_running and not self.stop_called:
                 self._timer = Timer(self.interval, self._run)
```

Comment:
I see you're still using `Timer`; any plan to integrate this with the existing thread pool, to avoid creating lots of threads?